src/tests/suite/daos_obj_ec.c

```diff
@@ -472,6 +472,169 @@ ec_partial_update_agg(void **state)
 	free(verify_data);
 }
 
+/* Submit a set of EC writes(defined by ec_ext), these writes should eventually
+ * compose one or N stripes, then wait for aggregation and verify these writes
+ * have been aggregated (no more replicated data in parity shard), also check
+ * if aggregated data are still correct.
+ */
+static void
+overlapped_update_agg(test_arg_t *arg, struct ec_ext *exts, int ext_nr)
+{
+	struct ioreq	req;
+	daos_recx_t	recx;
+	daos_obj_id_t	oid;
+	struct ec_ext	range = {0};
+	daos_off_t	data_off;
+	int		stripe_nr;
+	int		stripe_sz;
+	int		i;
+	char		*data;
+
+	if (!test_runable(arg, 6))
+		return;
+
+	daos_pool_set_prop(arg->pool.pool_uuid, "reclaim", "time");
+
+	/* Assuming object class is ec(2+1):
+	 * - these extents should have no gap (can overlap)
+	 * - they can compose N stripes (no misaligned header or tail).
+	 */
+	oid = daos_test_oid_gen(arg->coh, OC_EC_2P1G1, 0, 0, arg->myrank);
+	stripe_sz = EC_CELL_SIZE * 2;
+	/* convert all extents to the overall range */
+	ec_exts2range(exts, ext_nr, &range);
+
+	/* should be N stripes (no misaligned header or tail), so local aggregation
+	 * can actually do something.
+	 */
+	assert_true((range.e_start % stripe_sz == 0) && (range.e_end % stripe_sz == 0));
+
+	stripe_nr = (range.e_end - range.e_start) / stripe_sz;
+	/* at least one stripe */
+	assert_true(stripe_nr >= 1);
+
+	/* allocate buffer for the whole range */
+	data = (char *)malloc(stripe_nr * stripe_sz);
+	assert_true(data != NULL);
+
+	ioreq_init(&req, arg->coh, oid, DAOS_IOD_ARRAY, arg);
+
+	/* submit all the partially overlapped overwrites */
+	for (i = 0; i < ext_nr; i++) {
+		ec_ext2recx(&exts[i], &recx);
+		data_off = recx.rx_idx - range.e_start;
+		memset(&data[data_off], 'a' + i, recx.rx_nr);
+
+		req.iod_type = DAOS_IOD_ARRAY;
+		insert_recxs("d_key", "a_key", 1, DAOS_TX_NONE, &recx, 1,
+			     &data[data_off], recx.rx_nr, &req);
+	}
+
+	/* trigger EC aggregation on partiy shards */
+	trigger_and_wait_ec_aggreation(arg, &oid, 1, "d_key", "a_key", 0,
+				       stripe_sz * stripe_nr, DAOS_FORCE_EC_AGG);
+	for (i = 0; i < ext_nr; i++) {
+		ec_ext2recx(&exts[i], &recx);
+
+		data_off = recx.rx_idx - range.e_start;
+		ec_verify_parity_data(&req, "d_key", "a_key", recx.rx_idx, recx.rx_nr,
+				      &data[data_off], DAOS_TX_NONE, true);
+	}
+
+	ioreq_fini(&req);
+	free(data);
+}
+
+/* a set of overlapped extents, one full stripe (cell_size=1MB) */
```

Comment:
then the test needs to set the ec_cell_sz prop to be 1MB ?  in case we change the default cell size.